task_processing/plugins/kubernetes/task_config.py

```diff
@@ -37,21 +131,64 @@ def __invariant__(conf):
 
     uuid = field(type=str, initial=_generate_pod_suffix)  # type: ignore
     name = field(type=str, initial="default")
-    image = field(type=str, mandatory=True)
-    command = field(type=str,
-                    mandatory=True,
-                    invariant=lambda cmd: (cmd.strip() != '', 'empty command is not allowed'))
     node_selector = field(type=PMap)
     # Hardcoded for the time being
     restart_policy = "Never"
     # By default, the retrying executor retries 3 times. This task option
     # overrides the executor setting.
-    retries = field(type=int,
-                    factory=int,
-                    mandatory=False,
-                    invariant=lambda r: (r >= 0, 'retries >= 0'))
-    # TODO (TASKPROC-238): Add volume, cpu, gpu, desk, mem, etc.
-    volumes = field(type=PVector)
+    retries = field(
+        type=int,
+        factory=int,
+        mandatory=False,
+        invariant=lambda r: (r >= 0, 'retries >= 0')
+    )
+
+    image = field(type=str, mandatory=True)
+    command = field(
+        type=str,
+        mandatory=True,
+        invariant=lambda cmd: (cmd.strip() != '', 'empty command is not allowed')
+    )
+    volumes = field(
+        type=PVector if not TYPE_CHECKING else PVector["DockerVolume"],
+        initial=v(),
+        factory=pvector,
+        invariant=_valid_volumes,
+    )
+
+    cpus = field(
+        type=float,
+        initial=0.1,
+        factory=float,
+        invariant=lambda c: (c > 0, 'cpus > 0'))
+    memory = field(
+        type=float,
+        initial=128.0,
+        factory=float,
+        invariant=lambda m: (m >= 128, 'mem is >= 128'))
```

Comment:
Oh is this a kubernetes limitation, 128 is the minimum? We do have some jobs configured for less right now it seems.

Reply:
oh, I was just arbitrarily picking something "reasonable" since the Mesos plugin default seemed unreasonably low - but then i forgot to check what our existing values look like to see if the Mesos defaults still make sense :p